File: clients/src/main/java/org/yigitcanyontem/clients/content/dto/ReplyDto.java
package org.yigitcanyontem.clients.content.dto;

import lombok.*;
import org.yigitcanyontem.clients.users.dto.UsersDto;

import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ReplyDto {
    private Long id;

    private Long topicId;

    private TopicDto topic;

    private Integer createdByUserId;

    private String createdByUsername;

    private Date createdAt;

    private Date updatedAt;

    private Long parentReplyId;

    private String content;

    private Long upvoteCount;

    private Long downvoteCount;

    private Long childReplyCount;
}

File: frontend/codeConnect/src/models/content/ReplyDto.ts
import {TopicDto} from "@/models/content/TopicDto.ts";

export interface ReplyDto {
    id: number;
    topicId: number;
    topic: TopicDto;
    createdByUserId: number;
    createdByUsername: string;
    createdAt: Date;
    updatedAt: Date;
    parentReplyId: number;
    content: string;
    upvoteCount: number;
    downvoteCount: number;
    childReplyCount: number;
}
